app/views/admin/loans/timeline/_batch_actions.html.erb

```diff
@@ -12,14 +12,14 @@
   </li>
 
   <li>
-    <a id="adjust-dates" class="action" data-toggle="modal"
-      data-target="#adjust-dates-modal">
+    <a class="action adjust-dates" data-toggle="modal"
+      data-target="#<%= locals[:modal_id] %>">
```

Comment:
It's because you're using the short form of `render`, so you're actually passing `locals` as a hash. If you just change it to `render 'foo', modal_id: 123` it should work.